src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -475,9 +475,25 @@
       :else
       cols)))
 
+(defn- restore-cumulative-aggregations
+  [{aggregations :aggregation breakouts :breakout :as inner-query} replaced-indices]
+  (let [offset   (count breakouts)
+        restored (reduce (fn [aggregations index]
+                           (mbql.u/replace-in aggregations [(- index offset)]
+                             [:count]       [:cum-count]
+                             [:count field] [:cum-count field]
+                             [:sum field]   [:cum-sum field]))
+                         (vec aggregations)
+                         replaced-indices)]
+    (assoc inner-query :aggregation restored)))
+
 (defmethod column-info :query
-  [{inner-query :query} results]
-  (u/prog1 (mbql-cols inner-query results)
+  [{inner-query :query,
+    replaced-indices :metabase.query-processor.middleware.cumulative-aggregations/replaced-indecies}
```

Comment:
It's quite brittle to refer to data created by the `cumulative-aggregations` middleware here, but I couldn't find a simple way to encapsulate the whole functionality there, because post processing middleware cannot "modify" the query.